### PR TITLE
NIFI-5147 Calculate hash attribute redux

### PR DESCRIPTION
**Pull request by ottobackwards**
[#2802](https://github.com/apache/nifi/pull/2802)

The prior PR landed, but was reverted for further review.  I cannot re-open, so I'm creating a new PR.
Please reference the comments in the old pr.

### CODE REVIEW

**Comment by ottobackwards** [comment]
@alopresto, just a ping.  It has been a while

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestCalculateAttributeHash.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+        for (final MockFlowFile flowFile : success) {
+            Assert.assertEquals(Hex.encodeHexString(DigestUtils.getDigest(algorithm).digest("abcdefg".getBytes(UTF8))),
+                    flowFile.getAttribute(String.format("%s_%s", "name", algorithm)));
+            Assert.assertEquals(Hex.encodeHexString(DigestUtils.getDigest(algorithm).digest("hijklmnop".getBytes(UTF8))),
+                    flowFile.getAttribute(String.format("%s_%s", "value", algorithm)));
+        }
+    }
+
+    public void testParitalAlgorithm(String algorithm) {
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestCalculateAttributeHash.java]: Typo: `testPartialAlgorithm`

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestCalculateAttributeHash.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        final TestRunner runner = TestRunners.newTestRunner(new CalculateAttributeHash());
+        runner.setProperty(CalculateAttributeHash.HASH_ALGORITHM.getName(), algorithm);
+        runner.setProperty("name", String.format("%s_%s", "name", algorithm));
+        runner.setProperty("value", String.format("%s_%s", "value", algorithm));
+
+        // test default ALLOW
+
+        final Map<String, String> paritalAttributeMap = new HashMap<>();
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestCalculateAttributeHash.java]: Typo: `partialAttributeMap`

**Comment by alopresto** [comment]
I started reviewing this PR but realized it did not implement many of the behaviors I had needed in the original ticket. I opened [PR 2980](https://github.com/apache/nifi/pull/2980) which includes this foundation instead.

**Comment by ottobackwards** [comment]
Ok, that is great.  Did I miss something that _was_ in the jira?
Should I just close this PR now then?

**Comment by alopresto** [comment]
I didn't capture every detail of the issue in the Jira because I planned on writing it myself. However, some of the behavior (empty input should still return a hash, etc.) was standard. Yes, please close this PR and review 2980.

**Comment by ottobackwards** [comment]
Sure.  For Jira's when you want to do it yourself, I suggest assigning it.  That is my habit on the other Apache things I work with, and it will help guard against eager beavers ;)

Thanks for using the work as much as you did.

**Pull request closed without merging**